autoscheduler/frontend/src/tests/CourseSelectColumn.test.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+
+import { render, fireEvent, act } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import { createStore, applyMiddleware } from 'redux';
+import thunk from 'redux-thunk';
+import autoSchedulerReducer from '../redux/reducers';
+import CourseSelectColumn from '../components/CourseSelectColumn/CourseSelectColumn';
+
+describe('CourseSelectColumn', () => {
+  describe('Clicking the Add Course button', () => {
+    test('Adds a course card', () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+
+      const { getByText } = render(
+        <Provider store={store}>
+          <CourseSelectColumn />
+        </Provider>,
+      );
+
+      // act
+      // Press the button
+      act(() => { fireEvent.click(getByText('Add Course')); });
+
+      // Get the course cards
+      const cardsCount = document.getElementsByClassName('MuiPaper-root').length;
```

Comment:
React Testing Library discourages testing using CSS classes since users won't actually see those, and RTL encourages testing components as users will actually see/use them. Consequently, I recommend using some variant of `getAllByText` instead

Reply:
Yeah that makes total sense. Tbh I was struggling with how to select it in the first place and was just waiting on what you suggested, will fix.